neatnet/nodes.py

```diff
@@ -131,6 +132,60 @@ def _status(x: pd.Series) -> str:
     return "changed"
 
 
+def isolate_bowtie_nodes(edgelines: list | np.ndarray | gpd.GeoSeries) -> gpd.GeoSeries:
+    r"""
+    Bowties are rare edgecase whereby a component has:
+    * 2 unique nodes
+    * 2 edges that are loops
+    * 4 edges that have only 3 unique coord-pairs *after* sorting.
+
+        |\ /‾‾\ /|
+        | *    * |
+        |/ \__/ \|
+
+    Although extremely rare, these cases throw a wrench in the
+    efficient logic of ``get_components()``. See gh#214.
+    """
+
+    if isinstance(edgelines, list):
+        edgelines = np.array(edgelines)
+    if isinstance(edgelines, np.ndarray):
+        edgelines = gpd.GeoSeries(edgelines)
+    if isinstance(edgelines, gpd.GeoSeries):
+        edgelines = edgelines.to_frame("geometry")
```

Comment:
```suggestion
    if not isinstance(edgelines, gpd.GeoDataFrame):
        edgelines = gpd.GeoDataFrame(geometry=edgelines)
```

Same thing, no?

Reply:
Good point. Let me try locally to make sure tests still pass with that.

Reply:
Actually we can get rid of that condition altogether and simply have:

```python
edgelines = gpd.GeoDataFrame(geometry=edgelines)
```

Since input will always be `list | np.ndarray | gpd.GeoSeries`, no need for 

```python
if not isinstance(edgelines, gpd.GeoDataFrame):
```

Reply:
Do you concur?

Reply:
true